packages/transformer/src/test/standalone/IModelTransformerHub.test.ts

```diff
@@ -1662,6 +1662,122 @@ describe("IModelTransformerHub", () => {
+    if (IModelJsFs.existsSync(masterSeedFileName))
+      IModelJsFs.removeSync(masterSeedFileName);
+    // userlabel / codevalue: set json property named updateState on the physicalobject to value (simplified state updates)
+    const masterSeedState = {40:1, 2:2, 41:3, 42:4};
+    const masterSeedDb = SnapshotDb.createEmpty(masterSeedFileName, { rootSubject: { name: masterIModelName } });
+    masterSeedDb.nativeDb.setITwinId(iTwinId); // workaround for "ContextId was not properly setup in the checkpoint" issue
+    populateTimelineSeed(masterSeedDb, masterSeedState);
+
+    const noFedGuidElemIds = masterSeedDb.queryEntityIds({ from: "Bis.Element", where: "UserLabel IN (41, 42)" });
+    for (const elemId of noFedGuidElemIds)
+      masterSeedDb.withSqliteStatement(
+        `UPDATE bis_Element SET FederationGuid=NULL WHERE Id=${elemId}`,
+        (s) => { expect(s.step()).to.equal(DbResult.BE_SQLITE_DONE); }
+      );
+    masterSeedDb.performCheckpoint();
+
+    // hard to check this without closing the db...
+    const seedSecondConn = SnapshotDb.openFile(masterSeedDb.pathName);
+    for (const elemId of noFedGuidElemIds)
+      expect(seedSecondConn.elements.getElement(elemId).federationGuid).to.be.undefined;
+    seedSecondConn.close();
+
+    const masterSeed: TimelineIModelState = {
+      // HACK: we know this will only be used for seeding via its path and performCheckpoint
+      db: masterSeedDb as any as BriefcaseDb,
+      id: "master-seed",
+      state: masterSeedState,
+    };
+
+    const expectedRelationships = [
+      { sourceLabel: "40", targetLabel: "2", idInBranch: "not inserted yet", sourceFedGuid: true, targetFedGuid: true },
+      { sourceLabel: "41", targetLabel: "42", idInBranch: "not inserted yet", sourceFedGuid: false, targetFedGuid: false },
+    ];
+
+    let aspectId: Id64String | undefined;
+    const timeline: Timeline = [
+      { master: { seed: masterSeed } },
+      { branch: { branch: "master" } },
+      {
+        branch: {
+          manualUpdate(db) {
+            expectedRelationships.map(
+              ({ sourceLabel, targetLabel }, i) => {
+                const sourceId = IModelTestUtils.queryByUserLabel(db, sourceLabel);
+                const targetId = IModelTestUtils.queryByUserLabel(db, targetLabel);
+                assert(sourceId && targetId);
+                const rel = ElementGroupsMembers.create(db, sourceId, targetId, 0);
+                expectedRelationships[i].idInBranch = rel.insert();
+              }
+            );
+          },
+        },
+      },
+      { master: { sync: ["branch"] } }, // first master<-branch reverse sync
+      {
+        assert({branch}) {
+          // expectedRelationships[1] has no fedguids, so expect to find an esa.
+          const sourceId = IModelTestUtils.queryByUserLabel(branch.db, expectedRelationships[1].sourceLabel);
+          aspectId = branch.db.elements.getAspects(sourceId, ExternalSourceAspect.classFullName)[0].id;
```

Comment:
this should now have 2 ESAs, one for its own provenance, and one for the relationship. Can you assert that?

Reply:
I did it based off of the 'kind' of the ESA. Hopefully thats alright